src/shared/psi-util.c

```diff
@@ -0,0 +1,73 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "fileio.h"
+#include "psi-util.h"
+#include "stat-util.h"
+
+int read_resource_pressure(const char *path, PressureType type, ResourcePressure *ret) {
+        _cleanup_free_ char *value = NULL;
+        float avg10, avg60, avg300;
+        const char *field;
+        char *marker;
+        usec_t total;
+        char buf[6];
+        int r;
+
+        assert(path);
+        assert((type == PRESSURE_TYPE_SOME) || (type == PRESSURE_TYPE_FULL));
+        assert(ret);
+
+        if (type == PRESSURE_TYPE_SOME)
+                field = "some";
+        else if (type == PRESSURE_TYPE_FULL)
+                field = "full";
+
+        r = read_full_virtual_file(path, &value, NULL);
+        if (r < 0)
+                return r;
+
+        marker = value;
+
+        do {
+                r = sscanf(marker, "%5s avg10=%f avg60=%f avg300=%f total="USEC_FMT, buf, &avg10, &avg60, &avg300, &total);
+                if (r != 5)
```

Comment:
hmm, i am always a bit afraid to see new uses of sscanf(). I wonder if a extract_first_word() based loop wouldn't be better here... I mean, does the kernel even guarantee the order of the fields here? (if it did, it wouldn't have to have field names, hence my assumption is that the order can be changed anytime?).

i.e. something like this:

Reply:
I can try `extract_first_word()` instead.